File: part-02/src/main/java/eu/luminis/workshop/smallsteps/persistence/InMemoryLegoStockRepository.java
package eu.luminis.workshop.smallsteps.persistence;

import eu.luminis.workshop.smallsteps.logic.appservice.LegoStockRepository;
import eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects.LegoStoreId;
import eu.luminis.workshop.smallsteps.logic.domainservice.state.StockState;
import org.springframework.stereotype.Repository;

import java.util.HashMap;
import java.util.Map;

@Repository
public class InMemoryLegoStockRepository  implements LegoStockRepository {
    private final Map<LegoStoreId, StockState> data;

    public InMemoryLegoStockRepository() {
        data = new HashMap<>();
    }

    @Override
    public StockState find(LegoStoreId legoStoreId) {
        return data.getOrDefault(legoStoreId, new StockState());
    }

    @Override
    public void save(LegoStoreId legoStoreId, StockState newState) {
        data.put(legoStoreId, newState);
    }
}

File: part-02/src/main/kotlin/eu/luminis/workshop/smallsteps/persistence/InMemoryLegoStockRepository.kt
package eu.luminis.workshop.smallsteps.persistence

import eu.luminis.workshop.smallsteps.logic.appService.LegoStockRepository
import eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects.LegoStoreId
import eu.luminis.workshop.smallsteps.logic.domainService.state.StockState

class InMemoryLegoStockRepository : LegoStockRepository {
    private val data = mutableMapOf<LegoStoreId, StockState>()

    override fun find(store: LegoStoreId): StockState {
        return data[store] ?: StockState()
    }

    override fun save(store: LegoStoreId, newState: StockState) {
        data[store] = newState
    }
}

